src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,262 @@
+                case "delete": {
+                    String index = multiWord.next();
+                    int intIndex = Integer.parseInt(index);
+                    try {
+                        if (intIndex <= tl.getStorage().size() && intIndex > 0) {
+                            String returnString = tl.stringDelete(intIndex - 1);
+                            // save to save file
+                            stores.save(tl.getStorage());
+                            return returnString;
+                        } else {
+                            throw Ui.DukeException.outOfBounds(intIndex);
+                        }
+                    } catch (Ui.DukeException e) {
+                        return e.getMessage();
+                    }
+                }
+                case "find": {
+                    try {
+                        if (multiWord.hasNextLine()) {
+                            String remainingWords = multiWord.nextLine().trim();
+                            return stringFilterList(remainingWords, tl.getStorage());
+                        } else {
+                            throw Ui.DukeException.empty("find");
+                        }
+                    } catch (Ui.DukeException e) {
+                        return e.getMessage();
+                    }
+                }
+                case "due": {
+                    try {
+                        LocalDate dueDate = LocalDate.parse(multiWord.nextLine().trim());
+                        DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("MMM dd yyyy");
+                        StringBuilder response = new StringBuilder("These tasks are due: \n");
+                        for (Task task: tl.getDateStorage().get(dueDate.format(dateFormat))) {
+                            response.append(task).append("\n");
+                        }
+                        return response.toString();
+                    } catch (Exception e) {
+                        // time can't be parsed
+                        return e.getMessage();
+                    }
+                }
+                // it's a task
+                case "todo":
+                case "deadline":
+                case "event":
+                    try {
+                        // whitespace in front of nextLine
+                        if (multiWord.hasNextLine()) {
+                            String remainingWords = multiWord.nextLine().trim();
+                            String returnString = tl.stringStore(firstWord, remainingWords);
+                            // save to save file
+                            stores.save(tl.getStorage());
+                            return returnString;
+                        } else {
+                            throw Ui.DukeException.empty(firstWord);
+                        }
+                    } catch (Ui.DukeException e) {
+                        return e.getMessage();
+                    }
+                // invalid order
```

Comment:
Recommend you to create a different command classes for the various commands(i.e. DoneCommand, FindCommand etc) to better handle the user commands. This method is doing way too many things and is too long and convoluted